src/Controller/CRUDController.php

```diff
@@ -901,8 +914,11 @@ final public function configureAdmin(Request $request): void
             ));
         }
 
+        $pool = $this->get('sonata.admin.pool');
+        \assert($pool instanceof Pool);
+
         try {
-            $this->admin = $this->get('sonata.admin.pool')->getAdminByAdminCode($adminCode);
+            $this->admin = $pool->getAdminByAdminCode($adminCode);
```

Comment:
The issue will still be present in https://github.com/sonata-project/SonataAdminBundle/pull/7190

I think we don't have other option than adding a phpdoc.